synthesis/codon/codon.go

```diff
@@ -91,15 +91,39 @@ type Table interface {
 	GetAminoAcids() []AminoAcid
 	GetStartCodons() []string
 	GetStopCodons() []string
+	GetStats() *Stats
 	IsEmpty() bool
 	OptimizeTable(string) Table
 }
 
+// Stats denotes a set of computed codon table statistics
+type Stats struct {
+	StartCodonCount map[string]int `json:"start_codons_counts"`
```

Comment:
I agree with, instead of having a Stats struct that is added to the codon table, we have `ComputeStats()`. I think that is much cleaner. Although we'd still need to have `StartCodonCount`, since that can't be contained in the current struct.